tests/test_message_broker.py

```diff
@@ -0,0 +1,105 @@
+from catkit2.catkit_bindings import LocalMemory, MessageBroker, MessageSubscriptionMode
+import numpy as np
+import pytest
+
+@pytest.fixture(scope='module')
+def broker():
+    header = LocalMemory.create(1024 * 1024 * 512)
+    block = LocalMemory.create(1024 * 1024 * 1024)
+
+    broker = MessageBroker.create(header, [block])
+    yield broker
+
+def test_message_subscription(broker):
+    topic = "test_message_subscription"
+
+    subscription_newest = broker.subscribe(topic, mode=MessageSubscriptionMode.NewestOnly)
+    subscription_sequential = broker.subscribe(topic, mode=MessageSubscriptionMode.Sequential)
+
+    for i in range(3):
+        arr = np.array([i + 10]).astype('int32')
+        message = broker.prepare_message(topic, arr.nbytes)
+        message.payload = arr
+        broker.publish_message(message)
+
+    assert subscription_newest.next_message_id == 2
+    assert subscription_sequential.next_message_id == 0
+
+    message_1 = subscription_newest.get_next_message(0.01)
+    message_2 = subscription_sequential.get_next_message(0.01)
+    message_3 = subscription_sequential.get_next_message(0.01)
+
+    assert message_1 is not None
+    assert message_2 is not None
+    assert message_3 is not None
+
+    assert message_1.payload[0] == 12
+    assert message_2.payload[0] == 10
+    assert message_3.payload[0] == 11
+
+    assert subscription_newest.try_get_next_message() is None
+
+    m = subscription_sequential.try_get_next_message()
+    assert m is not None
+    assert m.payload[0] == 12
+
+    assert subscription_sequential.try_get_next_message() is None
+
+    subscription_newest2 = broker.subscribe(topic, starting_frame_id=1, mode=MessageSubscriptionMode.NewestOnly)
+    assert subscription_newest2.next_message_id == 2
```

Comment:
Why is this 2 if you give it a starting ID of 2? Does it not give you the same one? The sequential subscription below shows the behavior I expect, so why does the "newest" subscription mode automatically up the message id?

Reply:
`MessageSubscriptionMode.Sequential` will try to give you messages without dropped frames, even if there are newer messages available. `MessageSubscriptionMode.NewestOnly` will instead drop frames to give you the most recent update. In this case, `subscription_newest2` is trying to read frame 1, but sees that frame 2 is already there, so it will skip frame 1 and give you frame 2. `subscription_sequential2` will _not_ skip frames, and so will give you frame 1 first, then frame 2 afterwards (untested here, because that was already tested a few asserts above).

Reply:
(Potentially stupid) follow-up question: How come there is frame 2 already there?

Reply:
We submitted three frames to the message broker: frame 0, 1 and 2.

Reply:
Or I'm misunderstanding the question.

Reply:
Submitted where? I only see them submitted at the very top, and we access them all later on, so they shouldn't be there anymore, no? Or does a new subscription have access to all the messages?

Reply:
Messages are not consumed. You can access historical data as well. The last N messages are kept for each topic (currently N = 32), similar to how the last N frames are kept for datastreams.

Sorry, I should have said that earlier. This is different than Apacha Kafka or Aeron, if you're familiar with those. In those frameworks, messages are only delivered once, and only accessible if published after you created your subscription. In our message broker, messages are delivered once and then can be accessed later, kinda like a time series database. I'm not fully convinced about the latter capability, since I don't think we _ever_ used it on HiCAT. And it complicates the networking interface that I'm building now.